src/main/java/org/spectrumauctions/sats/mechanism/domains/ModelDomain.java

```diff
@@ -24,13 +53,26 @@
     private transient Allocation efficientAllocation;
 
     protected abstract ModelMIP getMIP();
-
+    
+    private boolean generic;
+    
+    @Setter
+    private int priceGenerationBidsPerBidder = 100;
+    @Setter
+    private int priceGenerationNumberOfWorldSamples = 100;
+    @Setter
+    private double priceGenerationFraction = 0.01;
+    @Setter
+    private long priceGenerationSeed = System.currentTimeMillis();
+    
+    
     public ModelDomain(List<? extends SATSBidder> bidders) {
         this(bidders, false);
     }
 
     public ModelDomain(List<? extends SATSBidder> bidders, boolean generic) {
         this.bidders = bidders;
+        this.generic = generic;
```

Comment:
I have troubles accepting the distinction between generic and non-generic here ;-) Because that's something we wanted to get rid of. Of course it's a straightforward change because it comes from the old SATS version, but I suggest the following:

Leave all the "distinguishable" parts to the children of ModelDomain. For example, below, you could have an abstract method called `getPricedGoods()` (or so), and the generic domain will implement this as world.getAllGenericDefinitions(), and the non-generic domains will implement this as world.getLicenses().
The same concept for the bid iterator below. I think that would be much more elegant, no?

Reply:
Of course you are right. But the existing implementation of the ModelDomain is already depending on the generic flag in the constructor. I did some refactoring to change that. I also eliminated some typecasts by using java generics in the ModelDomain.

Reply:
Amazing! 🙌